### ISSUE TITLE
Checkpoints given show low performance in baz and pmp tasks 

### ISSUE THREAD

**Issue by StarMoonWang** [senli1073/SeisT#2]
First of all, thanks for sharing the code of your amazing work in here.
But when I try to use the checkpoints you give to test performance, I noticed everything works fine except in baz and pmp tasks, ckpts for these 2 may have something wrong, could you please check again if you put wrong files in here by mistake? I'll appreciate that. 
![seist-l](https://github.com/user-attachments/assets/8d775ed8-f22c-4577-8fde-918c7905b85a)
![seist-m](https://github.com/user-attachments/assets/7006e603-5798-4eb5-8d2b-6117a394d4b1)
p1 for seist-l and p2 for seist-m results on DiTing, I used checkpoints trained on DiTing from this github repository.

**Comment by senli1073** [comment]
Thank you for pointing out this issue. I'll check it.

**Comment by senli1073** [comment]
Hi @Wang-XingHao,

I guess you accidentally missed some arguments (in Supporting Table S1). You could have a try with the following arguments.

- For pmp task: 
`--in-samples 256  --p-position-ratio 0.5` 

- For baz task:
`--in-samples 150  --p-position-ratio 0.33` 

This method refers to the implementation of DiTingMotion which fixes the phase-p position in the input seismogram to improve the model's accuracy.

**Comment by StarMoonWang** [comment]
Oh my bad, indeed I didn't notice this setting : (  now pmp performs well as expected, and baz is better but still low

> pmp_diting_light: (in-samples 256  p-position-ratio 0.5)
2024-09-24 16:41:43,652:`p_position_ratio` is 0.5, `shift_event_rate` -> `0.0`
2024-09-24 16:41:43,652:`p_position_ratio` is 0.5, `generate_noise_rate` -> `0.0`
2024-09-24 16:45:49,442:* [PMP]PRECISION 0.9782  RECALL 0.9783  F1 0.9783 
baz_diting_light: (--in-samples 150  --p-position-ratio 0.33)
2024-09-24 16:41:51,464:`p_position_ratio` is 0.33, `shift_event_rate` -> `0.0`
2024-09-24 16:41:51,464:`p_position_ratio` is 0.33, `generate_noise_rate` -> `0.0`
2024-09-24 16:50:40,757:* [BAZ]MEAN -1.1719  STD 81.5198  MAE 62.9514  R2 0.3764 
seed 2024:
2024-09-24 17:00:07,932:* [BAZ]MEAN -0.2894  STD 81.5949  MAE 62.9352  R2 0.3799 

my test script :
`python main.py \
    --seed 0 \
    --mode "test" \
    --model-name "seist_l_baz" \
    --checkpoint  "./logs/pretrained_seist_l_***_diting_light/checkpoints/model-baz.pth" \
    --log-base "./logs" \
    --log-step 600 \
    --data "/project/datasets/DiTing330km" \
    --dataset-name "diting_light" \
    --data-split true \
    --train-size 0.8 \
    --val-size 0.1 \
    --shuffle true \
    --workers 6 \
    --in-samples 150 \
    --p-position-ratio 0.33 \
    --batch-size 128 \`
   do I miss any other specific settings? thank you for kind reply again!

**Comment by senli1073** [comment]
From these args, it is not easy to locate the problem. Maybe you can contact me on WeChat (SKJ1073) for more convenient communication.

**Comment by StarMoonWang** [comment]
many thanks, I just found my co-worker trained a seist-l-baz model on diting_light from scratch quite long ago (train and test with normal input length=8192 and no specific p_position_ratio set) and it performs well with
> [BAZ]MEAN -0.0377  STD 68.0148  MAE 42.4609  R2 0.5659 

so I get no worries about this issue now, but still happy to communicate with you!
